content-api/content-actors/src/main/scala/org/sunbird/content/util/RetireManager.scala

```diff
@@ -0,0 +1,120 @@
+            throw new ClientException("ERR_INVALID_CONTENT_ID", "Please Provide Valid Content Identifier.")
+        }
+    }
+
+    private def updateNodesToRetire(request: Request)(implicit ec: ExecutionContext, oec: OntologyEngineContext): Future[Response] = {
+        RedisCache.delete(request.get("identifier").asInstanceOf[String])
+        val updateReq = new Request(request)
+        updateReq.put("identifiers", java.util.Arrays.asList(request.get("identifier").asInstanceOf[String], request.get("identifier").asInstanceOf[String] + HierarchyConstants.IMAGE_SUFFIX))
+        updateReq.put("metadata", new util.HashMap[String, AnyRef]() {
+            {
+                put(HierarchyConstants.STATUS, "Retired")
+                put(HierarchyConstants.LAST_UPDATED_ON, DateUtils.formatCurrentDate)
+                put(HierarchyConstants.LAST_STATUS_CHANGED_ON, DateUtils.formatCurrentDate)
+            }
+        })
+        DataNode.bulkUpdate(updateReq).map(node => ResponseHandler.OK())
+    }
+
+
+    private def handleCollectionToRetire(node: Node, request: Request)(implicit ec: ExecutionContext, oec: OntologyEngineContext): Future[Response] = {
+        if (StringUtils.equalsIgnoreCase("application/vnd.ekstep.content-collection", node.getMetadata.get("mimeType").asInstanceOf[String]) && finalStatus.contains(node.getMetadata.get("status"))) {
+            RedisCache.delete("hierarchy_" + node.getIdentifier)
+            val req = new Request(request)
+            req.getContext.put("schemaName", "collection")
+            req.put("identifier", request.get("identifier"))
+            ExternalPropsManager.fetchProps(req, List(HierarchyConstants.HIERARCHY)).map(resp => {
+                val hierarchyString = resp.getResult.toMap.getOrElse(HierarchyConstants.HIERARCHY, "").asInstanceOf[String]
+                if (StringUtils.isNotBlank(hierarchyString)) {
+                    val hierarchyMap = JsonUtils.deserialize(hierarchyString, classOf[util.HashMap[String, AnyRef]])
+                    val childIds = getChildrenIdentifiers(hierarchyMap)
+                    //TODO : Implement Delete Units from Elastic Search
+                    if (CollectionUtils.isNotEmpty(childIds)) {
+                        val topicName = Platform.getString("kafka.topics.graph.event", "sunbirddev.learning.graph.events")
+                        childIds.foreach(id => kfClient.send(ScalaJsonUtils.serialize(getLearningGraphEvent(request, id)), topicName))
+                        RedisCache.delete(childIds.map(id => "hierarchy_" + id): _*)
+                    }
+                    hierarchyMap.put("status", "Retired")
+                    hierarchyMap.put(HierarchyConstants.LAST_UPDATED_ON, DateUtils.formatCurrentDate)
+                    hierarchyMap.put(HierarchyConstants.LAST_STATUS_CHANGED_ON, DateUtils.formatCurrentDate)
+                    req.put("hierarchy", ScalaJsonUtils.serialize(hierarchyMap))
+                    ExternalPropsManager.saveProps(req)
```

Comment:
```suggestion
                    ExternalPropsManager.saveProps(req)
```
Instead of updating cassandra and neo4j seperatly, we can just cal DataNode.update. It will handle both the updates together.